### ISSUE TITLE
Modules

### ISSUE THREAD

**Issue by ianagbip1oti** [ianagbip1oti/ProC#17]
> optional: actual module system

 @InitializeSahib are you able to expand on what this means? e.g., what features should it have, or not have? Is there any language you think gets it close to right?

**Comment by initializesahib** [comment]
by this i mean you should be able to import/export pieces of code as "modules", to be reused later. another feature that could be added onto this is a package manager and repository so you can share your modules with other people.

some languages that do modules right are python, node.js, ruby, etc.
(i'm not saying these languages are good)

this is mainly to replace the C's convention of header files

**Comment by ianagbip1oti** [comment]
So, something like

```
import("my_cool_module")
```

that loads the functions defined in `my_cool_module.proc` is along the right rack?

**Comment by initializesahib** [comment]
yes, but it shouldn't be limited to just the current file

there could be a default search path (where the package manager installs them)

**Comment by ianagbip1oti** [comment]
Absolutely. The first step will be importing from a local project though. Package management is a whole other problem.

(And I've still got stuff to add before I add imports, like functions, haha)

**Comment by initializesahib** [comment]
obviously :)
